src/cargo/core/compiler/fingerprint/mod.rs

```diff
@@ -775,13 +775,26 @@ impl LocalFingerprint {
         key: K,
         envs: &BTreeMap<String, Option<OsString>>,
     ) -> LocalFingerprint {
-        let key = key.as_ref();
+        fn get_env_value(key: &str, envs: &BTreeMap<String, Option<OsString>>) -> Option<OsString> {
+            if cfg!(windows) {
+                let upper_case_key: String = key.to_uppercase();
+                for (k, v) in envs {
+                    if k.to_uppercase().eq(&upper_case_key) {
+                        return v.to_owned();
+                    }
+                }
+            }
+            env::var_os(key)
+        }
+
+        let key: &str = key.as_ref();
         let var = key.to_owned();
-        let val = envs
-            .get(key)
-            .map(|v| v.to_owned())
-            .or_else(|| Some(env::var_os(key)))
-            .and_then(|os_str| os_str?.into_string().ok());
+
+        let val: Option<String> = match envs.get(key) {
+            Some(v) => v.to_owned(),
+            None => get_env_value(key, envs),
+        }
```

Comment:
This feels like a strange split in logic
- Direct look up happens
- Call a function
  - Case insensitive lookup happens
  - actual env look up happens

Reply:
My previous thought was to try to keep the previous habit of all prioritizing finding case sensitive variables. Then look for case insensitive variables.

Maybe what I'm considering is redundant.

Reply:
Oh, I think that makes sense.  Its more the division of top-level code / function call felt odd.  We split the lookup into the map between the two calls and hide away the `std::env` look up in the case insensitive call (which has a generic, unclear name)